### PR TITLE
Make command-line arguments consistent with the env variables

### LINKED ISSUE #2653: Environment variables question
Why do the environment variables have different names than the command-line arguments (e.g., KOKKOS_NUM_DEVICES vs --kokkos-ndevices, KOKKOS_DEVICE_ID vs --kokkos-device)?  Wouldn't it be easier to maintain and for users to guess the right environment variable names if the arguments and environment variables shared the same names?

https://github.com/kokkos/kokkos/wiki/Initialization#53-initialization-by-environment-variables

Kokkos' environment variables are insufficiently documented.  I'm told @mhoemmen added the section above just yesterday, with just enough info to get me past a problem I was having.  (Thanks, @mhoemmen !)  But there are other initialization variables that are not documented, and users can't guess their names because the naming convention differs.

https://github.com/trilinos/Trilinos/pull/6590

### PR DESCRIPTION
**Pull request by dalg24**
Fix #2653

### CODE REVIEW

**Comment by dalg24** [comment]
Note: the [documentation](https://github.com/kokkos/kokkos/wiki/Initialization#51-initialization-by-command-line-arguments) after we merge

File: core/src/impl/Kokkos_Core.cpp [hunk 1]
```diff
@@ -474,42 +483,62 @@ void parse_command_line_arguments(int& narg, char* arg[],
...
+      if (check_arg(arg[iarg], "--ndevices")) {
+        warn_deprecated_command_line_argument("--ndevices", "--num-devices");
+      }
+      if (check_arg(arg[iarg], "--kokkos-ndevices")) {
+        warn_deprecated_command_line_argument("--kokkos-ndevices",
+                                              "--kokkos-num-devices");
+      }
       // Find the number of device (expecting --device=XX)
-      if (!((strncmp(arg[iarg], "--kokkos-ndevices=", 18) == 0) ||
+      if (!((strncmp(arg[iarg], "--kokkos-num-devices=", 21) == 0) ||
```
Comment (masterleinad) [inline core/src/impl/Kokkos_Core.cpp]: Any reason not to use `check_arg` here?

File: core/src/impl/Kokkos_Core.cpp [hunk 2]
```diff
@@ -552,88 +582,45 @@ void parse_command_line_arguments(int& narg, char* arg[],
...
-      std::cout << "-----------------------------------------------------------"
-                   "---------------------"
-                << std::endl;
-      std::cout << std::endl;
-
-      // Remove the --kokkos-help argument from the list but leave --ndevices
-      if (strcmp(arg[iarg], "--kokkos-help") == 0) {
+    } else if (check_arg(arg[iarg], "--kokkos-help") ||
+               check_arg(arg[iarg], "--help")) {
+      std::string const help_message = R"(
```
Comment (masterleinad) [inline core/src/impl/Kokkos_Core.cpp]:
```suggestion
      auto const help_message = R"(
```
would avoid the intermediate `std::string` to be a bit simpler.

File: core/src/impl/Kokkos_Core.cpp (hunk 1 above)
Comment (dalg24) [reply core/src/impl/Kokkos_Core.cpp]: I did not (and still don't) want to mess with what goes inside the statement-true of that outer if-else statement.  It has other problems but it is good enough and I don't want to rewrite it.

**Pull request merged**